cumulusci/robotframework/Salesforce.py

```diff
@@ -488,10 +489,58 @@ def salesforce_insert(self, obj_name, **kwargs):
         """ Inserts a Salesforce object setting fields using kwargs and returns the id """
         self.builtin.log("Inserting {} with values {}".format(obj_name, kwargs))
         obj_class = getattr(self.cumulusci.sf, obj_name)
+
         res = obj_class.create(kwargs)
         self.store_session_record(obj_name, res["id"])
+
         return res["id"]
 
+    def salesforce_collection_insert(self, objects):
+        """Inserts up to 200 records that were created with Salesforce Init Objects"""
+
+        def dict_to_insertable(d):
+            insertable = {"attributes": {"type": d["type"]}}
+            for key, value in d.items():
+                if key != "type":
+                    insertable[key] = value
+            return insertable
+
+        insertables = [dict_to_insertable(o) for o in objects]
+
+        return self.cumulusci.sf.restful(
+            "composite/sobjects",
+            method="POST",
+            json={"allOrNone": False, "records": insertables},
+        )
+
+    def salesforce_init_objects(self, obj_name, number_to_create, **fields):
+        """Create an array of dictionaries with template-formatted arguments appropriate for a Collection Insert.
+            Use ``{number}`` to represent the unique index of the row in the list of rows and ``{random_str}`` to represent a random string.
+            For example:
+
+                | @{objects} =  Salesforce Init Objects  Contact  3
+                | ...  FirstName="User {number}"
+                | ...  LastName="{random_str}"
+
+            Which would generate:
+
+                | [{'FirstName': '"User 0"', 'LastName': '"u4PNlGUD"', 'type': 'Contact'},
+                |  {'FirstName': '"User 1"', 'LastName': '"Kyd4PC5x"', 'type': 'Contact'},
+                |  {'FirstName': '"User 2"', 'LastName': '"n4EGs4Vp"', 'type': 'Contact'}]
+           """
+        objs = []
+        for i in range(int(number_to_create)):
+            obj = {"type": obj_name}  # Object type to create
```

Comment:
Opportunity has a standard field named `Type` so there may be some issues if someone tries to set that. (hi Paul! 👋)

Reply:
Welcome back!

Good point. Custom objects could have a "type" as well. Fixed here:

a18b92f4fa36997e31f490f5a940cdf19df040e7

Your feedback is always welcome!

Reply:
Clever solution!